test/e2e/packetcapture_test.go

```diff
@@ -694,44 +814,95 @@ func getOSString() string {
 	return "linux"
 }
 
-func runPacketCaptureTest(t *testing.T, data *TestData, tc pcTestCase) {
-	switch tc.ipVersion {
-	case 4:
-		skipIfNotIPv4Cluster(t)
-	case 6:
-		skipIfNotIPv6Cluster(t)
-	}
-
-	var dstPodIPs *PodIPs
-	if tc.pc.Spec.Destination.IP != nil {
-		ip := net.ParseIP(*tc.pc.Spec.Destination.IP)
+func getIPsForEndpoint(t *testing.T, data *TestData, podRef *crdv1alpha1.PodReference, ipStr *string) *PodIPs {
+	if ipStr != nil {
+		ip := net.ParseIP(*ipStr)
 		if ip.To4() != nil {
-			dstPodIPs = &PodIPs{IPv4: &ip}
-		} else {
-			dstPodIPs = &PodIPs{IPv6: &ip}
+			return &PodIPs{IPv4: &ip}
 		}
-	} else if tc.pc.Spec.Destination.Pod != nil {
-		pod, err := data.clientset.CoreV1().Pods(tc.pc.Spec.Destination.Pod.Namespace).Get(context.TODO(), tc.pc.Spec.Destination.Pod.Name, metav1.GetOptions{})
+		return &PodIPs{IPv6: &ip}
+	} else if podRef != nil {
+		pod, err := data.clientset.CoreV1().Pods(podRef.Namespace).Get(context.TODO(), podRef.Name, metav1.GetOptions{})
 		if err != nil {
 			require.True(t, errors.IsNotFound(err))
 		} else {
-			dstPodIPs, err = parsePodIPs(pod)
+			podIPs, err := parsePodIPs(pod)
 			require.NoError(t, err)
+			return podIPs
 		}
 	}
-	var srcPodIPs *PodIPs
-	if tc.pc.Spec.Source.IP != nil {
-		ip := net.ParseIP(*tc.pc.Spec.Source.IP)
-		srcPodIPs = &PodIPs{IPv4: &ip}
-	} else if tc.pc.Spec.Source.Pod != nil {
-		pod, err := data.clientset.CoreV1().Pods(tc.pc.Spec.Source.Pod.Namespace).Get(context.TODO(), tc.pc.Spec.Source.Pod.Name, metav1.GetOptions{})
-		if err != nil {
-			require.True(t, errors.IsNotFound(err))
-		} else {
-			srcPodIPs, err = parsePodIPs(pod)
-			require.NoError(t, err)
+	return nil
+}
+
+func determineSrcPod(tc pcTestCase, srcPodIPs *PodIPs) string {
+	if srcPodIPs != nil && tc.pc.Spec.Source.Pod != nil {
+		return tc.pc.Spec.Source.Pod.Name
+	}
+	return "client"
+}
```

Comment:
this function is very confusing IMO
first, it's unclear why `srcPodIPs` is even needed
second, it's unclear why we return `"client"` as the default value
maybe some comments would help